state/migration_export_test.go

```diff
@@ -858,6 +858,74 @@ func (s *MigrationExportSuite) TestLinkLayerDevicesSkipped(c *gc.C) {
 	c.Assert(devices, gc.HasLen, 0)
 }
 
+func (s *MigrationExportSuite) TestInstanceDataSkipped(c *gc.C) {
+	s.Factory.MakeMachine(c, &factory.MachineParams{
+		Constraints: constraints.MustParse("arch=amd64 mem=8G"),
+	})
+
+	model, err := s.State.ExportPartial(state.ExportConfig{
+		SkipInstanceData: true,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	listMachines := model.Machines()
+
+	instance := listMachines[0].Instance()
+	c.Assert(instance, gc.Equals, nil)
+}
+
+func (s *MigrationBaseSuite) TestMachineAgentBinariesSkipped(c *gc.C) {
+	s.Factory.MakeMachine(c, &factory.MachineParams{
+		Constraints: constraints.MustParse("arch=amd64 mem=8G"),
+	})
+
+	model, err := s.State.ExportPartial(state.ExportConfig{
+		SkipMachineAgentBinaries: true,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	listMachines := model.Machines()
+	tools := listMachines[0].Tools()
+	c.Assert(tools, gc.Equals, nil)
+}
+
+func (s *MigrationBaseSuite) TestUnitAgentBinariesSkipped(c *gc.C) {
+	dummyCharm := s.Factory.MakeCharm(c, &factory.CharmParams{Name: "dummy"})
+	application := s.Factory.MakeApplication(c, &factory.ApplicationParams{Name: "dummy", Charm: dummyCharm})
+
+	_, err := application.AddUnit(state.AddUnitParams{})
+	c.Assert(err, jc.ErrorIsNil)
+
+	model, err := s.State.ExportPartial(state.ExportConfig{
+		SkipUnitAgentBinaries: true,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	listApplications := model.Applications()
+	unit := listApplications[0].Units()
+	c.Assert(unit[0].Tools(), gc.Equals, nil)
+}
+
+func (s *MigrationBaseSuite) TestRelationScopeSkipped(c *gc.C) {
+	wordpress := state.AddTestingApplication(c, s.State, "wordpress", state.AddTestingCharm(c, s.State, "wordpress"))
+	mysql := state.AddTestingApplication(c, s.State, "mysql", state.AddTestingCharm(c, s.State, "mysql"))
+	// InferEndpoints will always return provider, requirer
+	eps, err := s.State.InferEndpoints("mysql", "wordpress")
+	c.Assert(err, jc.ErrorIsNil)
+	_, err = s.State.AddRelation(eps...)
+	c.Assert(err, jc.ErrorIsNil)
+	s.Factory.MakeUnit(c, &factory.UnitParams{Application: wordpress})
+	s.Factory.MakeUnit(c, &factory.UnitParams{Application: mysql})
+
+	model, err := s.State.ExportPartial(state.ExportConfig{
+		SkipRelationScope: true,
+		SkipSettings:      true,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	c.Assert(model.Relations(), gc.HasLen, 1)
```

Comment:
This test isn't really checking anything. Where is it removing the relation scopes to check that that no longer causes an error?

Reply:
Ignore me - the test does fail if SkipRelationScopes is false